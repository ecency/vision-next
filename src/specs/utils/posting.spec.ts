import {
  createPatch,
  createPermlink,
  createReplyPermlink,
  extractMetaData,
  makeCommentOptions,
  makeJsonMetaData,
  makeJsonMetaDataReply
} from "../../utils/posting";

describe("Posting", () => {
  it("createPermlink", () => {
    const input = "lorem ipsum dolor sit amet";
    expect(createPermlink(input)).toMatchSnapshot();
  });

  it("createPermlink random", () => {
    jest.spyOn(Math, "random").mockImplementation(() => {
      return 1.95136022969379;
    });
    const input = "lorem ipsum dolor sit amet";
    expect(createPermlink(input, true)).toMatchSnapshot();
  });

  it("createPermlink non-latin chars", () => {
    jest.spyOn(Math, "random").mockImplementation(() => {
      return 1.95136022969379;
    });
    const input = "ปลาตัวใหญ่สีเหลืองทอง";
    expect(createPermlink(input)).toMatchSnapshot();
  });

  it("(1) extractMetadata", () => {
    const input = '<img src="http://www.xx.com/a.png"> @lorem @ipsum';
    expect(extractMetaData(input)).toMatchSnapshot();
  });

  it("(2) extractMetadata", () => {
    const input =
      '@lorem <img src="http://www.xx.com/a.png"> ![h74zrad2fh.jpg](https://img.esteem.ws/h74zrad2fh.jpg) http://www.google.com/foo/bar  @ipsum';
    expect(extractMetaData(input)).toMatchSnapshot();
  });

  it("makeJsonMetaData", () => {
    const meta = {
      image: ["http://www.xx.com/a.png", "https://img.esteem.ws/h74zrad2fh.jpg"]
    };
    const tags = ["esteem", "art"];

    expect(makeJsonMetaData(meta, tags, "", "2.0.0")).toMatchSnapshot();
  });

  describe("makeCommentOptions", () => {
    it("(1) Default 50% / 50%", () => {
      expect(
        makeCommentOptions("talhasch", "lorem-ipsum-1", "default")
      ).toBeNull();
    });

    it("(2) Power Up 100%", () => {
      expect(makeCommentOptions("talhasch", "lorem-ipsum-1", "sp")).toMatchSnapshot();
    });

    it("(3) Decline Payout", () => {
      expect(makeCommentOptions("talhasch", "lorem-ipsum-1", "dp")).toMatchSnapshot();
    });

    it("(4) Empty beneficiary list", () => {
      expect(
        makeCommentOptions("talhasch", "lorem-ipsum-1", "default", [])
      ).toBeNull();
    });

    it("(5) With beneficiary list", () => {
      expect(
        makeCommentOptions("talhasch", "lorem-ipsum-1", "default", [
          { account: "foo", weight: 300 },
          { account: "bar", weight: 200 }
        ])
      ).toMatchSnapshot();
    });
  });

  it("makeJsonMetadataReply", () => {
    expect(makeJsonMetaDataReply(["foo", "bar"], "1.1")).toMatchSnapshot();
  });

  it("createReplyPermlink", () => {
    // @ts-ignore
    jest.spyOn(Date, "now").mockImplementation(() => {
      return new Date("2018-09-21T12:00:50.000Z");
    });
    expect(createReplyPermlink("good-karma")).toMatchSnapshot();
  });

  it("createPatch", () => {
    expect(
      createPatch("lorem ipsum dlor sit amet", "lorem ipsum dolor sit amet")
    ).toMatchSnapshot();
  });
});
