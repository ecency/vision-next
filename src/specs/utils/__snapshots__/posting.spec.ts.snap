// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posting (1) extractMetadata 1`] = `
{
  "image": [
    "http://www.xx.com/a.png",
  ],
  "thumbnails": [
    "http://www.xx.com/a.png",
  ],
}
`;

exports[`Posting (2) extractMetadata 1`] = `
{
  "image": [
    "http://www.xx.com/a.png",
    "https://img.esteem.ws/h74zrad2fh.jpg",
  ],
  "thumbnails": [
    "http://www.xx.com/a.png",
    "https://img.esteem.ws/h74zrad2fh.jpg",
  ],
}
`;

exports[`Posting createPatch 1`] = `
"@@ -6,16 +6,17 @@
  ipsum d
+o
 lor sit
"
`;

exports[`Posting createPermlink 1`] = `"lorem-ipsum-dolor-sit-amet"`;

exports[`Posting createPermlink non-latin chars 1`] = `"f38c58113ff7e"`;

exports[`Posting createPermlink random 1`] = `"lorem-ipsum-dolor-sit-amet-f38c58113ff7e"`;

exports[`Posting createReplyPermlink 1`] = `"re-good-karma-2018921t120500z"`;

exports[`Posting makeCommentOptions (2) Power Up 100% 1`] = `
{
  "allow_curation_rewards": true,
  "allow_votes": true,
  "author": "talhasch",
  "extensions": [],
  "max_accepted_payout": "1000000.000 HBD",
  "percent_hbd": 0,
  "permlink": "lorem-ipsum-1",
}
`;

exports[`Posting makeCommentOptions (3) Decline Payout 1`] = `
{
  "allow_curation_rewards": true,
  "allow_votes": true,
  "author": "talhasch",
  "extensions": [],
  "max_accepted_payout": "0.000 HBD",
  "percent_hbd": 10000,
  "permlink": "lorem-ipsum-1",
}
`;

exports[`Posting makeCommentOptions (5) With beneficiary list 1`] = `
{
  "allow_curation_rewards": true,
  "allow_votes": true,
  "author": "talhasch",
  "extensions": [
    [
      0,
      {
        "beneficiaries": [
          {
            "account": "bar",
            "weight": 200,
          },
          {
            "account": "foo",
            "weight": 300,
          },
        ],
      },
    ],
  ],
  "max_accepted_payout": "1000000.000 HBD",
  "percent_hbd": 10000,
  "permlink": "lorem-ipsum-1",
}
`;

exports[`Posting makeJsonMetaData 1`] = `
{
  "app": "ecency/2.0.0-vision",
  "description": "",
  "format": "markdown+html",
  "image": [
    "http://www.xx.com/a.png",
    "https://img.esteem.ws/h74zrad2fh.jpg",
  ],
  "tags": [
    "esteem",
    "art",
  ],
}
`;

exports[`Posting makeJsonMetadataReply 1`] = `
{
  "app": "ecency/1.1-vision",
  "format": "markdown+html",
  "tags": [
    "foo",
    "bar",
  ],
}
`;
